field/internal/templates/element/inverse_pornin20.go

```diff
@@ -0,0 +1,296 @@
+	// To make this constant-time-ish, replace the condition with i < invIterationsN
+	for i = 0; i&1 == 1 || !a.IsZero(); i++ {
+		n := max(a.BitLen(), b.BitLen())
+		aApprox, bApprox := approximate(&a, n), approximate(&b, n)
+
+		// After 0 iterations, we have f₀ ≤ 2⁰ and f₁ < 2⁰
+		f0, g0, f1, g1 = 1, 0, 0, 1
+
+		for j := 0; j < approxLowBitsN; j++ {
+
+			if aApprox&1 == 0 {
+				aApprox /= 2
+			} else {
+				s, borrow := bits.Sub64(aApprox, bApprox, 0)
+				if borrow == 1 {
+					s = bApprox - aApprox
+					bApprox = aApprox
+					f0, f1 = f1, f0
+					g0, g1 = g1, g0
+				}
+
+				aApprox = s / 2
+				f0 -= f1
+				g0 -= g1
+
+				// Now |f₀| < 2ʲ + 2ʲ = 2ʲ⁺¹
+				// |f₁| ≤ 2ʲ still
+			}
+
+			f1 *= 2
+			g1 *= 2
+			// |f₁| ≤ 2ʲ⁺¹
+		}
+
+		s = a
+		aHi := a.linearCombNonModular(&s, f0, &b, g0)
+		if aHi & signBitSelector != 0 {
+			// if aHi < 0
+			f0, g0 = -f0, -g0
+			aHi = a.neg(&a, aHi)
+		}
+		// right-shift a by k-1 bits
+
+		{{- range $i := .NbWordsIndexesFull}}
+			{{-  if eq $i $.NbWordsLastIndex}}
+				a[{{$i}}] = (a[{{$i}}] >> approxLowBitsN) | (aHi << approxHighBitsN)
+			{{-  else  }}
+				a[{{$i}}] = (a[{{$i}}] >> approxLowBitsN) | ((a[{{add $i 1}}]) << approxHighBitsN)
+			{{- end}}
+		{{- end}}
+
+		bHi := b.linearCombNonModular(&s, f1, &b, g1)
+		if bHi & signBitSelector != 0 {
+			// if bHi < 0
+			f1, g1 = -f1, -g1
+			bHi = b.neg(&b, bHi)
+		}
+		// right-shift b by k-1 bits
+
+		{{- range $i := .NbWordsIndexesFull}}
+			{{-  if eq $i $.NbWordsLastIndex}}
+				b[{{$i}}] = (b[{{$i}}] >> approxLowBitsN) | (bHi << approxHighBitsN)
+			{{-  else  }}
+				b[{{$i}}] = (b[{{$i}}] >> approxLowBitsN) | ((b[{{add $i 1}}]) << approxHighBitsN)
+			{{- end}}
+		{{- end}}
+
+		if i&1 == 1 {
+			// Combine current update factors with previously stored ones
+			// [f₀, g₀; f₁, g₁] ← [f₀, g₀; f₁, g₀] [pf₀, pg₀; pf₀, pg₀]
+			// We have |f₀|, |g₀|, |pf₀|, |pf₁| ≤ 2ᵏ⁻¹, and that |pf_i| < 2ᵏ⁻¹ for i ∈ {0, 1}
+			// Then for the new value we get |f₀| < 2ᵏ⁻¹ × 2ᵏ⁻¹ + 2ᵏ⁻¹ × 2ᵏ⁻¹ = 2²ᵏ⁻¹
+			// Which leaves us with an extra bit for the sign
+			f0, g0, f1, g1 = f0*pf0+g0*pf1,
+				f0*pg0+g0*pg1,
+				f1*pf0+g1*pf1,
+				f1*pg0+g1*pg1
+
+			s = u
+			u.linearCombSosSigned(&u, f0, &v, g0)
+			v.linearCombSosSigned(&s, f1, &v, g1)
+
+		} else {
+			// Save update factors
+			pf0, pg0, pf1, pg1 = f0, g0, f1, g1
+		}
+	}
+
+	// For every iteration that we miss, v is not being multiplied by 2²ᵏ⁻²
+	const pSq int64 = 1 << (2 * (k - 1))
+	// If the function is constant-time ish, this loop will not run (probably no need to take it out explicitly)
+	for ; i < invIterationsN; i += 2 {
+		v.mulWSigned(&v, pSq)
+	}
+
+	z.Mul(&v, &{{.ElementName}}{
+		{{- range $i := .NbWordsIndexesFull }}
+		inversionCorrectionFactorWord{{$i}},
+		{{- end}}
+	})
+	return z
+}
+
+// approximate a big number using its uppermost and lowermost bits
+func approximate(x *{{.ElementName}}, n int) uint64 {
+
+	if n <= 64 {
```

Comment:
question, just based on the function comment; if say, n == 24, shouldn't this clear the 40 other bits to 0?